Cython/Compiler/Dataclass.py

```diff
@@ -0,0 +1,629 @@
+            if py_name:
+                names = py_name.split(",")
+                names = [ n.strip("() ") for n in names ]
+            if names:
+                for name in names:
+                    name = EncodedString(name)
+                    nn = ExprNodes.NameNode(self.pos, name=name)
+                    # try to set the entry now to prevent the user accidentally shadowing
+                    # the name
+                    nn.entry = env.builtin_scope().lookup(name)
+                    if not nn.entry:
+                        try:
+                            nn.entry = env.declare_builtin(name, self.pos)
+                        except:
+                            pass  # not convinced a failure means much
+                    if nn.entry:
+                        return nn.analyse_types(env)
+
+        # otherwise we're left to return a string
+        s = self.entry.pep563_annotation
+        if not s:
+            s = self.entry.type.declaration_code("", for_display=1)
+        return ExprNodes.StringNode(self.pos, value=s).analyse_types(env)
+
+class DontAnalyseDeclarationsNode(ExprNodes.ExprNode):
+    # arg    ExprNode
+    #
+    # This is designed to wrap stuff that's already been analysed
+    # so that lambdas aren't redeclared for example
+    # and then immediately be replaced
+
+    subexprs = []
+
+    def analyse_declarations(self, env):
+        return
+
+class RemoveDontAnalyseDeclarations(VisitorTransform):
+    def visit_DontAnalyseDeclarationsNode(self, node):
+        return node.arg
+
+    def visit_Node(self, node):
+        self.visitchildren(node)
+        return node
+
+
+class FieldsValueNode(ExprNodes.ExprNode):
+    # largely just forwards arg. Allows it to be coerced to a Python object
+    # if possible, and if not then generates a sensible backup string
+    subexprs = ['arg']
+
+    def __init__(self, pos, arg):
+        super(FieldsValueNode, self).__init__(pos, arg=arg)
+
+    def analyse_types(self, env):
+        self.arg.analyse_types(env)
+        self.type = self.arg.type
+        return self
+
+    def coerce_to_pyobject(self, env):
+        if self.arg.type.can_coerce_to_pyobject(env):
+            return self.arg.coerce_to_pyobject(env)
+        else:
+            # A string representation of the code that gave the field seems like a reasonable
+            # fallback. This'll mostly happen for "default" and "default_factory" where the
+            # type may be a C-type that can't be converted to Python.
+            return self._make_string()
+
+    def _make_string(self):
+        from .AutoDocTransforms import AnnotationWriter
+        writer = AnnotationWriter(description="Dataclass field")
+        string = writer.write(self.arg)
+        return ExprNodes.StringNode(self.pos, value=EncodedString(string))
+
+    def generate_evaluation_code(self, code):
+        return self.arg.generate_evaluation_code(code)
+
+
+def _setup_dataclass_fields(node, fields, dataclass_module):
```

Comment:
`setup` is a noun. This function does something.
```suggestion
def _set_up_dataclass_fields(node, fields, dataclass_module):
```